--- ./crypto/compat/getentropy_linux.c.orig	2014-07-17 14:42:11.914193262 +0200
+++ ./crypto/compat/getentropy_linux.c	2014-07-17 14:42:39.316850682 +0200
@@ -486,6 +486,7 @@
 
 			HD(cnt);
 		}
+#ifdef HAVE_GETAUXVAL
 #ifdef AT_RANDOM
 		/* Not as random as you think but we take what we are given */
 		p = (char *) getauxval(AT_RANDOM);
@@ -502,6 +503,7 @@
 		if (p)
 			HD(p);
 #endif
+#endif
 
 		SHA512_Final(results, &ctx);
 		memcpy((char *)buf + i, results, min(sizeof(results), len - i));
